frontend/src/components/admin/UserCard.tsx

```diff
@@ -48,30 +52,32 @@ const UserCard = ({
   role,
   email,
   town,
+  onUserSelect,
 }: UserCardProps): React.ReactElement => {
-  const userCardOptions = ["User", "Admin", "Staff"];
-  const [selectValue, setSelectValue] = useState(userCardOptions[0]);
-  const [currRole, setCurrRole] = useState(role);
+  const [selectValue, setSelectValue] = useState("");
   const { isOpen, onOpen, onClose } = useDisclosure();
 
-  const [updateUser] = useMutation<{ updateUser: UserResponse }>(UPDATE_USER);
+  useEffect(() => {
+    setSelectValue(role);
+  }, [role]);
+
+  const [UpdateUserRole] = useMutation(UPDATE_USER_ROLE);
 
-  const resetState = () => {
-    setSelectValue(userCardOptions[0]);
+  const resetState = (userRole: Role) => {
+    setSelectValue(userRole);
   };
```

Comment:
```suggestion
  const resetState = setSelectValue;
```

Or just use `setSelectValue` directly.